lib/client/websocket-tracker.js

```diff
@@ -133,6 +148,18 @@ WebSocketTracker.prototype._onSocketData = function (data) {
     return
   }
 
+  if (data.action === common.ACTIONS.ANNOUNCE || data.offer || data.answer) {
+    self._onAnnounceResponse(data)
+  } else if (data.action === common.ACTIONS.SCRAPE) {
+    self._onScrapeResponse(data)
+  } else {
+    throw new Error('invalid action in WS response: ' + data.action)
```

Comment:
Just noticed this: Throwing is a bad idea. That means that any tracker that sends back an invalid response will cause an uncaught exception in the client. This should just call `self._onSocketError`